.github/workflows/build-lint-test.yml

```diff
@@ -79,20 +78,14 @@ jobs:
     runs-on: ubuntu-latest
     needs: prepare
     steps:
-      - uses: actions/checkout@v4
-      - name: Setup Node
-        uses: actions/setup-node@v4
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
         with:
-          node-version-file: '.nvmrc'
-          cache: yarn
-      - run: yarn --immutable --immutable-cache
+          is-high-risk-environment: false
       - name: Restore "@metamask/snaps-execution-environments" build
-        uses: actions/cache@v4
+        uses: actions/download-artifact@v4
         with:
-          path: |
-            packages/snaps-execution-environments/dist/browserify
-          key: snaps-execution-environments-build-${{ runner.os }}-18.x-${{ github.sha }}
-          fail-on-cache-miss: true
+          name: snaps-execution-environments-build-${{ runner.os }}-22.x-${{ github.sha }}
```

Comment:
Does this fail on cache miss?

Reply:
Looks like it yes:

https://github.com/actions/download-artifact/blob/cc203385981b70ca67e1cc392babf9cc229d5806/src/download-artifact.ts#L68-L70